torch/distributed/_device_mesh.py

```diff
@@ -228,13 +228,12 @@ def _validate_mesh(self):
                 f"DeviceMesh cannot have duplicate values, but found {self.mesh.tolist()}"
             )
 
-        # validate that all calling ranks pass in the same `mesh` argument.
+        # # validate that all calling ranks pass in the same `mesh` argument.
         self_mesh = self.mesh.to(self.device_type).contiguous()
-        mesh_tensor = funcol.all_gather_tensor(
-            self_mesh, gather_dim=0, group=_get_default_group()
-        )
-        mesh_tensor_chunked = torch.chunk(mesh_tensor, get_world_size())
-        for other_rank, other_mesh in enumerate(mesh_tensor_chunked):
+        tensor_list = [torch.zeros_like(self_mesh) for _ in range(get_world_size())]
+        all_gather(tensor_list, self_mesh)
```

Comment:
Is there a reason against using `dist.all_gather_into_tensor()`? The extra copies from `dist.all_gather()` should be negligible for the input sizes, so it should not be a big deal either way.

Reply:
> Is there a reason against using `dist.all_gather_into_tensor()`? The extra copies from `dist.all_gather()` should be negligible for the input sizes, o it should not be a big deal either way.

No particular reason for preference here, but I decided to use all_gather() since I don't have to chunk the big tensor into a tensor list afterwards. But as you said, either way we should be fine because of the small input size.